std/container/array.d

```diff
@@ -385,9 +385,17 @@ if (!is(immutable T == immutable bool))
         if (isImplicitlyConvertible!(Elem, T))
         {
             import core.lifetime : emplace;
+            assert(_capacity >= length);
             if (_capacity == length)
             {
-                reserve(1 + capacity * 3 / 2);
+                import core.checkedint : addu;
+
+                bool overflow;
+                immutable size_t newCapacity = addu(capacity, capacity / 2 + 1, overflow);
+                if (overflow)
+                    assert(false, "Overflow");
+
+                reserve(newCapacity);
```

Comment:
Single-element `insertBack` prior to this PR:

https://github.com/dlang/phobos/blob/61c00065a8c7efdf21efb92ce69d7a9de4904ec9/std/container/array.d#L384-L397

If there is an overflow on line 390 `reserve` will be called with an argument smaller than the current `capacity`. After this the assertion on line 392 will fail with the message `""Failed to reserve memory"` because the call did not increase `capacity`. So overflow is already being detected.

Reply:
In `array.d`, other checks for "Overflow" (which are less serious than this one -- checks after addition, compared to this one which is check after scaling) are enabled even in `-release` mode.

The "Failed to reserve memory" check is not.